File: csharpcore/GildedRose.UnitTests/GildedRoseTest.cs
using System.Collections.Generic;
using System.Linq;
using GildedRoseKata;
using Xunit;

namespace GildedRose.UnitTests;

public class GildedRoseTest
{
    private static IList<Item> CreateItems() => new List<Item>{
        new() {Name = "Bananas", SellIn = 10, Quality = 20},
        new() {Name = "Aged Brie", SellIn = 2, Quality = 1},
        new() {Name = "Eggs", SellIn = 5, Quality = 7},
        new() {Name = "Eggs", SellIn = 12, Quality = 5},
        new() {Name = "Canned Beans", SellIn = 0, Quality = 80},
        new() {Name = "Canned Beans", SellIn = -1, Quality = 80},
        new() {Name = "Baked Sourdough Bread", SellIn = 3, Quality = 5}
    };

    private static readonly IList<Item> DefaultItems = CreateItems();
    private static readonly IList<Item> NewItems = CreateItems();
    
    private readonly GildedRoseKata.GildedRose _gildedRose = new(NewItems);
    
    [Fact]
    public void ShouldNotModifyCannedBeans()
    {
        var oldBeans = DefaultItems.Where(i => i.Name == "Canned Beans");
        
        _gildedRose.UpdateQuality();

        var newBeans = NewItems.Where(i => i.Name == "Canned Beans");
        
        
        Assert.All(newBeans, bean =>
        {
           Assert.Equal(80, bean.Quality); 
           Assert.Contains(oldBeans, o => o.SellIn == bean.SellIn);
        });
    }
}

File: js/gilded_rose.test.js
const chai = require('chai');
const sinon = require('sinon');
const expect = chai.expect;
const { Item, Shop } = require('./src/gilded_rose.js');


const items = [
  new Item("Bananas", 10, 20),
  new Item("Aged Brie", 2, 0),
  new Item("Eggs", 5, 7),
  new Item("Eggs", 12, 5),
  new Item("Canned Beans", 0, 80),
  new Item("Canned Beans", -1, 80),

  // This Baked good does not work properly yet!
  new Item("Baked Sourdough Bread", 3, 1),
];

describe('Gilded Rose', () => {

  describe('the updateQuality method', () => {

    it('updates quality', () => {
      console.log(items, 'items');
      const shop = new Shop(items);
      const shopsItems = shop.items;
      shop.updateQuality();

      expect(shopsItems[0].name).to.equal('Bananas');
      expect(shopsItems[0].sellIn).to.equal(9);
      expect(shopsItems[0].quality).to.equal(19);

      expect(shopsItems[1].quality).to.equal(1);
      expect(shopsItems[4].quality).to.equal(80);

      expect(shopsItems[6].quality).to.equal(0);
    })
  })
})


